runtime/slice.go

```diff
@@ -156,6 +156,58 @@ func initSliceType(map[string]*Object) {
 	SliceType.slots.Repr = &unaryOpSlot{sliceRepr}
 }
 
+func sliceIndex(f *Frame, o *Object) (i int64, raised *BaseException) {
```

Comment:
Since CPython handles it, I think we should handle it as well. One relatively simple way to do it for *Long values would be to do maxIntBig.Cmp(start.Value()) and always return -1 if the pos param is bigger than maxIntBig.

How does CPython implement this check?